src/components/wallet/methods/claimAsset.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  Account,
+  TransactionBuilder,
+  BASE_FEE,
+  Networks,
+  Operation,
+  Keypair,
+} from 'stellar-sdk'
+import { handleError } from '@services/error'
+import { Wallet } from '../wallet'
+
+export default async function claimAsset(
+  this: Wallet,
+  balanceId: string,
+  assetCode: string,
+  sponsor: string,
+  amount: string
+) {
+  const loadingKey = `claim:${assetCode}:${balanceId}`
+  this.loading = { ...this.loading, [loadingKey]: true }
+  const finish = () => (this.loading = { ...this.loading, [loadingKey]: false })
```

Comment:
Interesting approach. Why does the key need to be an array of size one? And where is this key-value pair used?